### PR TITLE
Add roundtrip test case for null buffer test

### PR DESCRIPTION
**Pull request by Jefffrey**
As suggested in this comment: https://github.com/datafusion-contrib/orc-rust/pull/13#pullrequestreview-2441984667

Instead of relying on PyORC to generate a file which tests the fix, we can rely on our own writing behaviour to generate a test file in the expected format.

### CODE REVIEW

File: src/arrow_writer.rs [hunk 1]
```diff
@@ -474,4 +475,62 @@ mod tests {
...
+        assert!(present_stream.is_some());
+
+        // Decoding from bytes
+        let reader = builder.build();
+        let rows = reader.collect::<Result<Vec<_>, _>>().unwrap();
+
+        assert_eq!(rows.len(), 1);
+        assert_eq!(rows[0].num_columns(), 1);
+        // Ensure read array has no null buffer
+        assert!(rows[0].column(0).nulls().is_none());
```
Comment (Jefffrey) [inline src/arrow_writer.rs:534]: A bit verbose but has important checks to ensure that if our ORC writer behaviour changes (such that it no longer writes an empty present stream) we'll be informed of this test breaking

**Review by Jefffrey — COMMENTED** [review]
Thoughts on this @progval ?

**Comment by progval** [comment]
thanks!
